### DISCUSSION TITLE
Extracting name and variables from GraphQL post request body

### DISCUSSION THREAD

**Discussion by svoruganti** [ChilliCream/hotchocolate#5767] [Q&A]
Hi,

Using ASP.Net core middleware, I am capturing GraphQL post request body as string. I would like to extract name and variables from the request string. Are there any built in classes that would allow me to do that. For example, if I have a string representation of following request:
```graphql
mutation createUser {
  user(
    input: {
      name: "test name"
      emailAddress: "[test@testing.com](mailto:test@testing.com)"
    }
  ) {
    id
  }
}
```
From the above request string, I would like to extract '**user**' and variables **name** and **emailAddress** along with values. We can extract the data using HotChocolate request middleware, but I want to find out if it is possible to do with http request string/stream i.e., without using HotChocolate request middleware.

**Comment by PascalSenn** [comment]
Yes you can use `Utf8GraphQLParser.Parse` to parse GraphQL string into a `DocumentNode`. Then you have the AST and can read the different parts easily
